rholang/src/main/scala/coop/rchain/rholang/interpreter/Runtime.scala

```diff
@@ -86,19 +109,28 @@ object Runtime {
     val errorLog                                  = new ErrorLog()
     implicit val ft: FunctorTell[Task, Throwable] = errorLog
 
-    def dispatchTableCreator(
-        space: RhoISpace,
-        dispatcher: Dispatch[Task, ListChannelWithRandom, TaggedContinuation]) = Map(
-      0L -> SystemProcesses.stdout,
-      1L -> SystemProcesses.stdoutAck(space, dispatcher),
-      2L -> SystemProcesses.stderr,
-      3L -> SystemProcesses.stderrAck(space, dispatcher),
-      4L -> SystemProcesses.ed25519Verify(space, dispatcher),
-      5L -> SystemProcesses.sha256Hash(space, dispatcher),
-      6L -> SystemProcesses.keccak256Hash(space, dispatcher),
-      7L -> SystemProcesses.blake2b256Hash(space, dispatcher),
-      9L -> SystemProcesses.secp256k1Verify(space, dispatcher)
-    )
+    def dispatchTableCreator(space: RhoISpace, dispatcher: RhoDispatch): RhoDispatchMap = {
+      import BodyRefs._
+      val registry = new Registry(space, dispatcher)
```

Comment:
`space` is not `PureRSpace` so it will throw as @ArturGajowy pointed out some time ago. This has some advantage like not having to deal with an effect (`F[_]`) when  doing `install`s in the `Registry`.

Reply:
Although, the truth is that we use `ISpace` to install `SystemProcesses` as well.

Reply:
No you don't pass the `PureRSPace` in the runtime. `space` is of type `RhoISpace` is just `ISpace`.